matrix/matrix.py

```diff
@@ -0,0 +1,189 @@
+    implementation only supports even-side matrices (n * n).
+
+    Attributes:
+        _matrix (tuple): contains the data about the matrix. Each line of
+                         the matrix is represented by a tuple.
+    """
+    __slots__ = '_matrix'
+
+    def __init__(self, *lines):
+        self._validate(*lines)
+        super(Matrix, self).__setattr__('_matrix', tuple(lines))
+
+    def __setattr__(self, key, value):
+        """Defines that Matrix attributes are immutable."""
+        raise AttributeError(f'{self.__class__} instance cannot be modified')
+
+    @staticmethod
+    def _validate(*data: tuple):
+        """Checks if the data given is valid. Valid data is defined if it
+        contains tuples of the same length (Matrix only supports n * n) and
+        their value is an integer."""
+        # check if each line is a tuple.
+        if not all([isinstance(line_data, tuple) for line_data in data]):
+            raise ValueError("Matrix lines must be tuple")
+
+        # check if each item in each line is an int or float.
+        if not all([any([isinstance(item, int), isinstance(item, float)]) for
+                    line_data in data for item in line_data]):
+            raise ValueError("Matrix items must be int or float")
+
+        # checks if the given data is of a n * n matrix.
+        if not all([len(line) == len(data) for line in data]):
+            raise ValueError("Matrix must be even-sided")
+
+    @classmethod
+    def unity(cls, number_lines: int):
+        """Generates a matrix with a main diagonal of 1's, everything else
+        is 0's.
+        """
+        tuples_in_chars = [
+            list(str('1' + ('0' * i)).zfill(number_lines)[::-1]) for i in
+            range(number_lines)]
+
+        return cls(*tuple(
+            [tuple(map(int, tuple_chars)) for tuple_chars in tuples_in_chars]))
+
+    @classmethod
+    def ones(cls, number_lines):
+        """Generates a matrix full of 1's."""
+        return cls(*tuple(
+            [tuple(repeat(1, number_lines)) for _ in range(number_lines)]))
+
+    def __repr__(self):
+        """Returns a string presentation of the matrix instance."""
+        return f"({', '.join([str(line) for line in self._matrix])})"
+
+    @property
+    def tuples(self):
+        """Returns the matrix instance as a tuple of tuples."""
+        return self._matrix
+
+    @staticmethod
+    def _bool_operation(operation, data) -> bool:
+        """Checks boolean operation output.
+
+        Args:
+            operation (function): the boolean function to operate.
+            data (list): the data of the matrix (list of tuples).
+
+        Returns:
+            If the operation is 'equal', returns true if all items are equal
+            to their parallel items in other, else false. If operation is
+            'not equal', return true if one or more items are not equal to
+            their parallel item/s in other, else false.
```

Comment:
first you should write the type that returned.
You can write it in much simpler way (for example, check if matrices equal element-wise)